spyql/interactive.py

```diff
@@ -0,0 +1,97 @@
+import os
+
+from .cli import clean_query, parse
+from .processor import Processor
+from .log import *
+
+class Q:
+  def __init__(self, query: str) -> None:
+    """
+    Make spyql interactive.
+
+    [ IMPORT python_module [ AS identifier ] [, ...] ]
+    SELECT [ DISTINCT | PARTIALS ]
+        [ * | python_expression [ AS output_column_name ] [, ...] ]
+        [ FROM csv | spy | text | python_expression | json [ EXPLODE path ] ]
+        [ WHERE python_expression ]
+        [ GROUP BY output_column_number | python_expression  [, ...] ]
+        [ ORDER BY output_column_number | python_expression
+            [ ASC | DESC ] [ NULLS { FIRST | LAST } ] [, ...] ]
+        [ LIMIT row_count ]
+        [ OFFSET num_rows_to_skip ]
+        [ TO csv | json | spy | sql | pretty | plot ]
+
+    Usage
+    -----
+
+    .. code-block:: python
+
+      >>> q = Q("IMPORT numpy SELECT numpy.mean(data->salary) FROM data WHERE data->name == 'akash'")
+      >>> q(data = data)
+
+    Args
+    ----
+
+      query(str): SpyQL string
+      input_opt/output_opt: kwargs for the input and writers, in this case of interactive mode we can
+        ignore these
+    """
+    self.query = query
+    self.parsed, self.strings = parse(clean_query(query))
+    self.output_path = None
+
+    # from logic has to be improved to include the files, python datastructures
```

Comment:
I have added the check in `cli.run`, since the requirements by CLI and Interactive can differ (python objects can be passed in Interactive) adding the logic in parse did not make sense.